addons/mail/models/discuss/ir_websocket.py

```diff
@@ -80,17 +120,81 @@ def _build_bus_channel_list(self, channels):
+
+        :param typing.List[str] channels: List of channels to subscribe to sent
+            by the client.
+        :param int last: Last known notification sent by the client.
+
+        :return:
+            A dict containing the following keys:
+            - channels (set of str): The list of channels to subscribe to.
+            - last (int): The last known notification id.
+            - missed_presences (odoo.models.Recordset): The missed presences.
+
+        :raise ValueError: If the list of channels is not a list of strings.
+        """
+        if not all(isinstance(c, str) for c in channels):
+            raise ValueError("bus.Bus only string channels are allowed.")
+        # sudo - bus.bus: reading non-sensitive last bus id.
+        last = 0 if last > self.env["bus.bus"].sudo()._bus_last_id() else last
+        str_presence_channels = {
+            c for c in channels if isinstance(c, str) and c.startswith("odoo-presence-")
+        }
+        presence_channels = self._build_presence_channel_list(
+            [tuple(c.replace("odoo-presence-", "").split("_")) for c in str_presence_channels]
+        )
+        # There is a gap between a subscription client side (which is debounced)
+        # and the actual subcription thus presences can be missed. Send a
+        # notification to avoid missing presences during a subscription.
+        domain = expression.AND(
+            [
+                [("last_poll", ">", datetime.now() - timedelta(seconds=2))],
+                expression.OR(self._get_missed_presences_identity_domains(presence_channels)),
+            ]
+        )
+        # sudo: mail.presence: can access presences linked to presence channels.
+        missed_presences = self.env["mail.presence"].sudo().search(domain)
+        all_channels = OrderedSet(presence_channels)
+        all_channels.update(
+            self._build_bus_channel_list([c for c in channels if c not in str_presence_channels])
+        )
+        return {"channels": all_channels, "last": last, "missed_presences": missed_presences}
+
+    def _subscribe(self, og_data):
+        data = self._prepare_subscribe_data(og_data["channels"], og_data["last"])
+        dispatch.subscribe(data["channels"], data["last"], self.env.registry.db_name, wsrequest.ws)
+        if bus_target := self._get_missed_presences_bus_target():
+            data["missed_presences"]._send_presence(bus_target=bus_target)
+        return super()._subscribe(og_data)
 
     @add_guest_to_context
     def _update_bus_presence(self, inactivity_period, im_status_ids_by_model):
-        super()._update_bus_presence(inactivity_period, im_status_ids_by_model)
+        if self.env.user and not self.env.user._is_public():
+            self.env['mail.presence'].update_presence(
+                inactivity_period,
+                identity_field='user_id',
+                identity_value=self.env.uid
+            )
         if not self.env.user or self.env.user._is_public():
```

Comment:
I think `else` would achieve the same result. Also, since both branches lead to calling `_update_presence`, I believe we could simplify this.